NwPluginAPI/Core/Extensions/ItemExtensions.cs

```diff
@@ -0,0 +1,220 @@
+		public static uint GetRandomAttachments(this ItemType firearm)
+		{
+			if (!firearm.IsFirearm())
+				throw new ArgumentException("The provided argument is not a valid firearm.", nameof(firearm));
+
+			return AttachmentsUtils.GetRandomAttachmentsCode(firearm);
+		}
+
+		/// <summary>
+		/// Applies the specified attachments to the given <see cref="Firearm"/>.
+		/// </summary>
+		/// <param name="firearm">The <see cref="Firearm"/> to which the attachments will be applied.</param>
+		/// <param name="attachmentCode">The attachments code representing the set of attachments to apply.</param>
+		public static void ApllyAttachments(this Firearm firearm, uint attachmentCode)
+		{
+			if (firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
+
+			firearm.ApplyAttachmentsCode(attachmentCode, true);
+
+			FirearmStatusFlags firearmStatusFlags = FirearmStatusFlags.MagazineInserted;
+			if (firearm.HasAdvantageFlag(AttachmentDescriptiveAdvantages.Flashlight))
+			{
+				firearmStatusFlags |= FirearmStatusFlags.FlashlightEnabled;
+			}
+			firearm.Status = new(firearm.AmmoManagerModule.MaxAmmo, firearmStatusFlags, firearm.GetCurrentAttachmentsCode());
+		}
+
+		/// <summary>
+		/// Applies a set of random attachments to the specified <see cref="Firearm"/>.
+		/// </summary>
+		/// <param name="firearm">The <see cref="Firearm"/> to which random attachments will be applied.</param>
+		/// <exception cref="ArgumentNullException">Thrown if the <paramref name="firearm"/> is null.</exception>
+		public static void ApllyRandomAttachments(this Firearm firearm)
+		{
+			if (firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
+
+			firearm.ApplyAttachmentsCode(AttachmentsUtils.GetRandomAttachmentsCode(firearm.ItemTypeId), true);
+
+			FirearmStatusFlags firearmStatusFlags = FirearmStatusFlags.MagazineInserted;
+			if (firearm.HasAdvantageFlag(AttachmentDescriptiveAdvantages.Flashlight))
+			{
+				firearmStatusFlags |= FirearmStatusFlags.FlashlightEnabled;
+			}
+			firearm.Status = new(firearm.AmmoManagerModule.MaxAmmo, firearmStatusFlags, firearm.GetCurrentAttachmentsCode());
+		}
+
+		/// <summary>
+		/// Applies the preferences of a <see cref="Player"/>'s attachments to a given <see cref="Firearm"/>.
+		/// </summary>
+		/// <param name="firearm">The <see cref="Firearm"/> to modify.</param>
+		/// <param name="player">The <see cref="Player"/> whose attachments preferences are to be applied.</param>
+		/// <param name="reloadWeapon">If this value is true, the weapon will be fully reloaded with ammunition when the attachment is applied.</param>
+		/// <returns>Returns true if the weapon was modified with the attachments; otherwise, false.</returns>
+		public static bool TryApplyAttachmentsPreferences(this Firearm firearm, Player player, bool reloadWeapon = false)
+		{
+			if(firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
```

Comment:
Perhaps returning false here could be better, as it is a "Try" style method

Reply:
makes sense I will do it